### PR TITLE
RLI pneumatics example code

### LINKED ISSUE #2165: Fix Pneumatic hardware API example code
The code block [here](https://docs.wpilib.org/en/stable/docs/software/hardware-apis/pneumatics/pneumatics.html#generating-and-storing-pressure) shows `getCompressorCurrent()` rather than `getCurrent()`

Also, that page should be more clear about the purpose of that code block- users shouldn't just copy and paste that code into their program.

### PR DESCRIPTION
**Pull request by Starlight220**
Fixes #2165

### CODE REVIEW

File: source/docs/software/hardware-apis/pneumatics/pneumatics.rst [hunk 1]
```diff
@@ -32,35 +32,47 @@ CAN Devices are identified by their Node ID. The default Node ID for PCMs is 0.
...
...t, but must be powered by the robot's pneumatics module. The "Closed Loop" mode on the Compressor is enabled by default, and it is *not* recommended that teams change this setting. When closed loop control is enabled the pneumatic module will automatically turn the compressor on when the digital pressure switch is closed (below the pressure threshold) and turn it off when the pressure switch is open (~120PSI). When closed loop control is disabled the compressor will not be turned on. Using the ``Compressor`` (`Java <https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/Compressor.html>`__ / `C++ <https://github.wpilib.org/allwpilib/docs/release/cpp/classfrc_1_1_compressor.html>`__) class, users can query the status of the compressor. The state (currently on or off), pressure switch state, and compressor current can all be queried from the Compressor object, as shown by the following code from the Solenoid example project (`Java <https://github.com/wpilibsuite/allwpilib/tree/main/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/solenoid>`__, `C++ <https://github.com/wpilibsuite/allwpilib/tree/main/wpilibcExamples/src/main/cpp/examples/Solenoid>`__):
```
Comment (sciencewhiz) [inline source/docs/software/hardware-apis/pneumatics/pneumatics.rst:35]: Isn't an off robot compressor illegal now?

Comment (sciencewhiz) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst]: This still talks about an off robot compressor

Comment (sciencewhiz) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst]: See also #2176

Comment (sciencewhiz) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst]: Never mind, github was outdated

File: source/docs/software/hardware-apis/pneumatics/pneumatics.rst [hunk 2]
```diff
@@ -32,35 +32,47 @@ CAN Devices are identified by their Node ID. The default Node ID for PCMs is 0.
...
-        frc::Compressor pcmCompressor{0, frc::PneumaticsModuleType::CTREPCM};
-        frc::Compressor phCompressor{1, frc::PneumaticsModuleType::REVPH};
-
-        pcmCompressor.EnableDigital();
-        pcmCompressor.Disable();
-
-        bool enabled = pcmCompressor.Enabled();
-        bool pressureSwitch = pcmCompressor.GetPressureSwitchValue();
-        double current = pcmCompressor.GetCompressorCurrent();
+    .. group-tab:: Java
```
[the remark this answers is not in the document]
Comment (Starlight220) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst:41]: The same applies to both, and I think that it is intuitive and specified somewhere to replace `REVPH` with `CTREPCM`.

[the remark this answers is not in the document]
Comment (sciencewhiz) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst:41]: Where? I don't see it

[the remark this answers is not in the document]
Comment (Starlight220) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst:41]: So it's maybe not specified, but I think it is intuitive.

[the remark this answers is not in the document]
Comment (Daltz333) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst:41]: I disagree, it should be mentioned or included.

[the remark this answers is not in the document]
Comment (Starlight220) [reply source/docs/software/hardware-apis/pneumatics/pneumatics.rst]: Fixed.
The code samples are now split, making it (hopefully) clear what is device-specific and what applies to both.

File: source/docs/software/hardware-apis/pneumatics/index.rst [hunk 3]
```diff
@@ -4,4 +4,30 @@ Pneumatics APIs
 .. toctree::
    :maxdepth: 1
 
-   pneumatics
+   solenoids
+   pressure
+
+Using the FRC Control System to control Pneumatics
```
Comment (Daltz333) [inline source/docs/software/hardware-apis/pneumatics/index.rst:10]:
```suggestion
Using the FRC Control System to Control Pneumatics
```

File: source/docs/software/hardware-apis/pneumatics/index.rst [hunk 4]
```diff
@@ -4,4 +4,30 @@ Pneumatics APIs
...
+The CTRE Pneumatics Control Module (PCM) is a CAN-based device that provides control over the compressor and up to 8 solenoids per module.
+
+.. image:: /docs/controls-overviews/images/control-system-hardware/pneumatic-hub.png
+    :alt: The Pneumatic Hub (PH)
+    :width: 400
+
+The REV Pneumatic Hub (PH) is a CAN-based device that provides control over the compressor and up to 16 solenoids per module.
+
+These devices are integrated into WPILib through a series of classes that make them simple to use. The closed loop control of the Compressor and Pressure switch is handled by the PCM hardware and the Solenoids are handled by the ``Solenoid`` class that controls the solenoid channels.
+These modules are responsible for regulating the robot's pressure using a pressure switch and a compressor and switching solenoids on and off. They communicate with the roboRIO over CAN. For more information, see :doc:`/docs/controls-overviews/control-system-hardware`.
```
Comment (Daltz333) [inline source/docs/software/hardware-apis/pneumatics/index.rst:28]: Newline separating paragraphs

File: source/docs/software/hardware-apis/pneumatics/pressure.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+.. note:: The Compressor object is only needed if you want the ability to turn off the compressor, change the pressure sensor (PH only), or query compressor status.
```
Comment (Daltz333) [inline source/docs/software/hardware-apis/pneumatics/pressure.rst:6]: For REV, this is how you enable/disable the switchable port.

File: source/docs/software/hardware-apis/pneumatics/solenoids.rst [hunk 6]
```diff
@@ -0,0 +1,117 @@
+.. include:: <isonum.txt>
+
+Operating pneumatic cylinders
```
Comment (Daltz333) [inline source/docs/software/hardware-apis/pneumatics/solenoids.rst:3]: Use capital case

File: source/docs/software/hardware-apis/pneumatics/solenoids.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Single acting solenoids apply or vent pressure from a single output port. They are typically used either when an external force will provide the return action of the cylinder (spring, gravity, separate mechanism) or in pairs to act as a double solenoid. A double solenoid switches air flow between two output ports (many also have a center position where neither output is vented or connected to the input). Double solenoid valves are commonly used when you wish to control both the extend and retract actions of a cylinder using air pressure. Double solenoid valves have two electrical inputs which connect back to two separate channels on the solenoid breakout.
```
Comment (Daltz333) [inline source/docs/software/hardware-apis/pneumatics/solenoids.rst:8]: A picture of both would be useful here

**Review by Daltz333 — CHANGES_REQUESTED** [review]

File: source/docs/software/hardware-apis/pneumatics/pressure.rst (hunk 5 above)
Comment (Starlight220) [reply source/docs/software/hardware-apis/pneumatics/pressure.rst:6]: What switchable port? This is the PH, not the PDH.

File: source/docs/software/hardware-apis/pneumatics/solenoids.rst (hunk 7 above)
Comment (Starlight220) [reply source/docs/software/hardware-apis/pneumatics/solenoids.rst:8]: I'm not sure where to bring pictures from.

**Review by sciencewhiz — APPROVED** [review]

**Comment by Starlight220** [comment]
@Daltz333 can you re-review this? Is there something blocking this?

**Comment by sciencewhiz** [comment]
This depends on wpilibsuite/allwpilib#4998 right?

**Comment by Starlight220** [comment]
I expect that the two are delaying each other. However, you're right. The other should be merged first, and I'll update this one.

**Comment by sciencewhiz** [comment]
needs merge conflict fixed, and I think the commit hashes should be updated now that it's been merged in allwpilib

**Review by sciencewhiz — APPROVED** [review]

**Comment by Starlight220** [comment]
@Daltz333 is this good to go?

**Pull request merged**